### ISSUE TITLE
`reportUnecessaryContains` not triggered on `dict_keys`

### ISSUE THREAD

**Issue by aidandj** [microsoft/pyright#11148]
**Describe the bug**
A contains check of a well typed dict `keys` method with an invalid argument does not trigger `reportUnecessaryContains`

**Code or Screenshots**
Code sample in [pyright playground](https://pyright-play.net/?strict=true&reportUnnecessaryComparison=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySAzjADRQAiSAxuVCAKYBujAhgDYD68CjAUPw4AuQiRgBtVDAC6UALxQJARjIAmMgGYZgpMCgAWTCigj%2BUC1AQhpACgDkACyT2AlLv0AiA5%2BOnh5pbWdk4u7vwAJqLUdFIo9NJyigDeysLKAL6CTKycPIiMthEAdADWjHDEtq7ueoZ%2BJeWV1QGWVjbxDs5uHlDevqhQjRVVrq1BHTBdYUA)

```python
from typing import List, Dict, reveal_type

l: List[int] = [1,2,3]

if 4 in l:
    print('hi')

if "4" in l: # Expression will always evaluate to False since the types "Literal['4']" and "int" have no overlap  (reportUnnecessaryContains)
    print('hi')

d: Dict[int, int] = {1:1}

reveal_type(d.keys()) 
if 4 in d.keys(): 
    print('hi')

if "4" in d.keys(): # Expect error here
    print('hi')
```

**VS Code extension or command-line**
1.1.407

**Comment by erictraut** [comment]
The `reportUnnecessaryContains` works only for a limited set of built-in types: `list`, `set`, `frozenset`, `deque`, `tuple`, `dict`, `defaultdict`, `OrderedDict`. It doesn't work with arbitrary container types.

**Comment by aidandj** [comment]
Bummer, thanks!

For future people finding this issue, this was bad code anyway. A normal `x in dict` call is better than `x in dict.keys()`, faster, ~and unnecessary contains works as well.~
